### ISSUE TITLE
case records - break data validation for backdating/correcting

### ISSUE THREAD

**Issue by HansVraga** [USGS-WiM/CBRADMS#102]
From User feedback:
>We need to be able to enter data in the CBRS Unit, Map Number, Map Date, and prohibition date fields as well as select from the dropdown lists.   The dropdowns make it so you are picking from valid effective data; however, sometimes we need to back date or correct older cases.

**Comment by HansVraga** [comment]
I intend to ignore this feedback until next round of review. Asking for the ability to break the very data validations they ask for sounds like a next phase enhancement - not a core requirement.

**Comment by HansVraga** [comment]
@aaronstephenson - you enabled backdating (with warning) as part of #101 - correct? If so, we can close!

**Comment by aaronstephenson** [comment]
This is a completely separate issue. Here, they're asking for the ability to override the static select list values and add their own. I am very hesitant to allow this, since those values come directly from their published maps.

**Comment by aaronstephenson** [comment]
Another *huge* issue with this request is that it breaks our current data model. Right now, CBRS unit, map number, and map date are all related to each other with foreign keys. What they are requesting is to abandon relations and store the data they enter directly in the case table, essentially denormalizing the DB.

**Comment by HansVraga** [comment]
No idea what I was thinking here: ☝️ https://github.com/USGS-WiM/CBRADMS/issues/102#issuecomment-287741905

**Comment by HansVraga** [comment]
We talked about breaking normalization for these fields in the past - Since CBRS unit, map number, and map dates can change, but they want to store the old values with the records (not have them update) we talked about breaking normalization to track this. 

Looked up the email thread: 
Sent: 16Sep16 
Subject: Follow-up on the Schema changes - Update
> All,
> Paul, Aaron, and I met to review the schema/data plans, and we stumbled on a CBRS process we hadn't accounted for. Specifically, we should expect a (roughly) 2 time a year update to the active list of: Maps (Panels), Units (Polygons), Prohibition Dates (Prohibitions). 
> 
> We thought the updates to these were less frequent (or extensive).
> 
> It's good Paul got this in front of us, we think we have a solution ready that will account for this and not be too complex.
> 
> First, we'll document the process to export the tables (and required columns, and relations table) from the geodatabase to a format that can be imported into the CBRS Determinations Database. Then, we'll document the process for importing that information. Once documented, we think this will be something Paul will work with ECOS to accomplish (as we won't be able to push the changes to the system once it's live in ECOS).
> 
> For saving the old data for determinations that are completed, we're going to make some changes to our schema to allow for saving of values (rather than index numbers) to the determinations table once a determination is complete. That way, even if we update the tables above with new active values, the old values will remain.
> 
> This will take a bit of time (hence the update), but should save a significant amount of future headaches...
> 
> Everyone have a good weekend!
>

**Comment by aaronstephenson** [comment]
Ah yes, thanks for the reminder. ugh. OK, I'll get working on this.

**Comment by HansVraga** [comment]
With the addition of the Legacy Map/Unit combo widget and agreement that we should handle Legacy Map/Units within the database, is this still necessary?

**Comment by aaronstephenson** [comment]
The decision made today (to implement a widget to allow the user to add a map and date on demand) is essentially what they asked for initially. We aren't going to abandon the database table relations; instead, we're just going to allow them to add records to the map table as needed. There may also be a need in the future to do bulk additions of old/deprecated data, but they didn't commit to anything on the phone call.

**Comment by HansVraga** [comment]
Thanks! That is how I understand the situation as well. Closing this one!
